panic at SpinnerLogSink.AddValues() and potentially logger.WithValues()

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
For logger method WithValues() and SpinnerLogSink.AddValues() the values array is not checked nor allocated prior to attempting to assign to specific indices.  This causes a panic that looks like:

```
panic: runtime error: index out of range [0] with length 0[0m[39m
goroutine 1 [running]:[0m[39m
github.com/banzaicloud/operator-tools/pkg/logger.(*SpinnerLogSink).AddValues(...)[0m[39m
/home/ubuntu/go/pkg/mod/github.com/banzaicloud/operator-tools@v0.28.3/pkg/logger/logsink.go:214
```

**Steps to reproduce the issue:**
<!-- Please describe the steps to reproduce the issue. -->
Use of SpinnerLogSink.AddValues() with non-zero len array arg.

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Don't crash

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Additional context**
<!-- Add any other context about the problem like release version, branch, etc. -->
v0.28.3
master

--- a/pkg/logger/logger.go
+++ b/pkg/logger/logger.go
@@ -85,9 +85,8 @@ func (log *logger) WithName(name string) Logger {
 
 func (log *logger) WithValues(keysAndValues ...interface{}) Logger {
 	sink := log.sink.copyLogger()
-	for k, v := range keysAndValues {
-		sink.values[k] = v
-	}
+
+	sink.values = keysAndValues
 
 	return &logger{
 		level: log.level,

--- a/pkg/logger/logsink.go
+++ b/pkg/logger/logsink.go
@@ -223,9 +223,7 @@ func (log *SpinnerLogSink) GetValues() []interface{} {
 }
 
 func (log *SpinnerLogSink) AddValues(keyAndValues []interface{}) {
-	for k, v := range keyAndValues {
-		log.values[k] = v
-	}
+	log.values = append(log.values, keyAndValues...)
 }
 
 func (log *SpinnerLogSink) AddName(name string) {

